### PR TITLE
Add check to skip folder when loading files

### PR DESCRIPTION
**Pull request by sunank200**
# Description
## What is the current behavior?

When load_file is passed patterns paths of S3 the operator is falling.

closes: #733 

## What is the new behavior?

- Check for folder while resolving the path pattern
- Add tests for the same.

### Checklist
- [x] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: src/astro/files/base.py [hunk 1]
```diff
@@ -142,15 +144,21 @@ def resolve_file_path_pattern(
...
+        if path.endswith("/"):
+            continue
+
+        files.append(
+            File(
+                path=path,
+                conn_id=conn_id,
+                filetype=filetype,
+                normalize_config=normalize_config,
+            )
```
Comment (kaxil) [inline src/astro/files/base.py:160]:
```suggestion

    files = [
        File(
            path=path,
            conn_id=conn_id,
            filetype=filetype,
            normalize_config=normalize_config,
        )
        for path in location.paths if not path.endswith("/")
    ]
```

Comment (kaxil) [reply src/astro/files/base.py:160]: Logically same but uses list comprehension

Comment (sunank200) [reply src/astro/files/base.py]: Ohh, I also added the same change there: https://github.com/astronomer/astro-sdk/pull/740/commits/124c42f624b6d7867a3e37c94a81335b04a67d5a. Sorry for that.

File: src/astro/files/base.py [hunk 2]
```diff
@@ -38,11 +40,11 @@ def __init__(
         )
 
     @property
-    def path(self) -> str:
+    def path(self) -> Any:
```
Comment (kaxil) [inline src/astro/files/base.py:43]: Why do we need to change this?

File: src/astro/files/base.py [hunk 3]
```diff
@@ -38,11 +40,11 @@ def __init__(
...
         return self.location.path
 
     @property
-    def conn_id(self) -> str | None:
+    def conn_id(self) -> Any:
```
Comment (kaxil) [inline src/astro/files/base.py:47]: Why do we need to change this?

File: tests/databases/test_snowflake.py [hunk 4]
```diff
@@ -200,8 +200,8 @@ def test_if_exist_param_of__load_pandas_dataframe_to_table(database_table_fixtur
 def test_load_file_to_table(database_table_fixture):
     """Test loading on files to snowflake database"""
     database, target_table = database_table_fixture
-    filepath = str(pathlib.Path(CWD.parent, "data/sample.csv"))
-    database.load_file_to_table(File(filepath), target_table, {})
+    filepath = str(pathlib.Path(CWD.parent, "data/sub_folder/"))
```
Comment (kaxil) [inline tests/databases/test_snowflake.py:203]: We need an integration test for checking this behaviour for remote storage like S3 or GCS too

File: src/astro/files/base.py (hunk 2 above)
Comment (sunank200) [reply src/astro/files/base.py:43]: Surprisingly, I had not changed that file but its throwing the following mypy error:
```
src/astro/files/base.py: note: In member "path" of class "File":
src/astro/files/base.py:44: error: Returning Any from function declared to
return "str"  [no-any-return]
```

But the type of path is `str`

File: src/astro/files/base.py (hunk 3 above)
Comment (sunank200) [reply src/astro/files/base.py:47]: I had not changed this code but getting this mypy error:
```
src/astro/files/base.py: note: In member "conn_id" of class "File":
src/astro/files/base.py:48: error: Returning Any from function declared to
return "Optional[str]"  [no-any-return]
```

Return type is `Optional[str]` which is equivalent to `str | None`

File: tests/databases/test_snowflake.py (hunk 4 above)
Comment (sunank200) [reply tests/databases/test_snowflake.py:203]: Added it here: https://github.com/astronomer/astro-sdk/pull/740/commits/544a21cb4982af2184337c4e461c258715b8c397

I have uploaded the similar folder structure on s3 bucket

File: src/astro/files/base.py (hunk 3 above)
Comment (kaxil) [reply src/astro/files/base.py]: The test passed when I reverted the changes @sunank200 in https://github.com/astronomer/astro-sdk/pull/740/commits/96a17d6bb035db24ee99d68bfbb09af9da01816a 

Where was it failing for you?

File: src/astro/files/base.py (hunk 2 above)
Comment (kaxil) [reply src/astro/files/base.py]: The test passed when I reverted the changes @sunank200 in https://github.com/astronomer/astro-sdk/pull/740/commits/96a17d6bb035db24ee99d68bfbb09af9da01816a 

Where was it failing for you?

**Review by kaxil — APPROVED** [review]

File: src/astro/files/base.py [hunk 5]
```diff
@@ -40,11 +39,11 @@ def __init__(
         )
 
     @property
-    def path(self) -> Any:
+    def path(self) -> str:
```
Comment (sunank200) [inline src/astro/files/base.py:42]: @kaxil does this work for you? Its throwing mypy error for me on my local

File: src/astro/files/base.py (hunk 2 above)
Comment (sunank200) [reply src/astro/files/base.py]: It failed for me in local while doing the commit @kaxil

File: src/astro/files/base.py (hunk 5 above)
Comment (kaxil) [reply src/astro/files/base.py:42]: Works for me

**Pull request merged**